packages/core/admin/server/controllers/license.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const fs = require('fs');
+// eslint-disable-next-line node/no-extraneous-require
+const ee = require('@strapi/strapi/lib/utils/ee');
+
+module.exports = {
+  async findLicense(ctx) {
+    let licenseInfo = {};
+
+    if (strapi.EE) {
+      licenseInfo = ee.licenseInfo;
+    } else {
+      licenseInfo = null;
+    }
+
+    // Make the license field disabled if not in development env
+    if (process.env.NODE_ENV !== 'development') {
+      return (ctx.body = {
+        data: {
+          messageId: 'Settings.license.wrong-env',
+          disabled: true,
+          licenseInfo,
+        },
+      });
+    }
+
+    // License file path
+    const licenseFilePath = './license.txt';
+
+    // Default messageId & activationType
+    let messageId = 'Settings.license.no-license';
+
+    if (process.env.STRAPI_LICENSE) {
+      messageId = 'Settings.license.strapi-license.found';
+
+      if (process.env.STRAPI_LICENSE == '') {
+        messageId = 'Settings.license.strapi-license.empty';
+      }
+    }
+
+    try {
+      // If the file already exists
+      if (fs.existsSync(licenseFilePath)) {
+        messageId = 'Settings.license.license-file.found';
+
+        const data = fs.readFileSync(licenseFilePath, 'utf8');
```

Comment:
Similar to my comment below: a license might be set through the `STRAPI_LICENSE` environment variable already, which this should take into account I think.